nibabel/cifti2/cifti2_axes.py

```diff
@@ -0,0 +1,1430 @@
+        other : Scalar
+            scalar axis to be compared
+
+        Returns
+        -------
+        bool : False if type, length or content do not match
+        """
+        if not isinstance(other, Scalar) or self.size != other.size:
+            return False
+        return np.array_equal(self.name, other.name) and np.array_equal(self.meta, other.meta)
+
+    def __add__(self, other):
+        """
+        Concatenates two Scalars
+
+        Parameters
+        ----------
+        other : Scalar
+            scalar axis to be appended to the current one
+
+        Returns
+        -------
+        Scalar
+        """
+        if not isinstance(other, Scalar):
+            return NotImplemented
+        return Scalar(
+                np.append(self.name, other.name),
+                np.append(self.meta, other.meta),
+        )
+
+    def __getitem__(self, item):
+        if isinstance(item, integer_types):
+            return self.get_element(item)
+        return self.__class__(self.name[item], self.meta[item])
+
+    def get_element(self, index):
+        """
+        Describes a single element from the axis
+
+        Parameters
+        ----------
+        index : int
+            Indexes the row/column of interest
+
+        Returns
+        -------
+        tuple with 2 elements
+        - unicode name of the row/column
+        - dictionary with the element metadata
+        """
+        return self.name[index], self.meta[index]
+
+
+class Label(Axis):
+    """
+    Defines CIfTI2 axis for label array.
+
+    Along this axis of the CIfTI2 vector/matrix each row/column has been given a unique name,
+    label table, and optionally metadata
+    """
+
+    def __init__(self, name, label, meta=None):
+        """
+        Parameters
+        ----------
+        name : np.ndarray
+            (N, ) string array with the parcel names
+        label : np.ndarray
+            single dictionary or (N, ) object array with dictionaries mapping
+            from integers to (name, (R, G, B, A)), where name is a string and R, G, B, and A are
+            floats between 0 and 1 giving the colour and alpha (i.e., transparency)
+        meta :  np.ndarray
+            (N, ) object array with a dictionary of metadata for each row/column
+        """
+        self.name = np.asanyarray(name, dtype='U')
+        if isinstance(label, dict):
+            label = [label.copy() for _ in range(self.name.size)]
+        self.label = np.asanyarray(label, dtype='object')
+        if meta is None:
+            meta = [{} for _ in range(self.name.size)]
+        self.meta = np.asanyarray(meta, dtype='object')
+
+        for check_name in ('name', 'meta', 'label'):
+            if getattr(self, check_name).shape != (self.size, ):
+                raise ValueError("Input {} has incorrect shape ({}) for Label axis".format(
+                        check_name, getattr(self, check_name).shape))
+
+    @classmethod
+    def from_mapping(cls, mim):
```

Comment:
I'd rather use `index_mapping` instead of `mapping`. This is because `mapping` makes me think of the `Mapping` collection and I try to use it with a `dict` object or similar.

Reply:
done